### ISSUE TITLE
`AdvancedHMCExt` will use `evaluate!!` to rerun the model for every sample drawn, and it seems wasteful  

### ISSUE THREAD

**Issue by sunxd3** [TuringLang/JuliaBUGS.jl#94]
See conversation https://github.com/TuringLang/JuliaBUGS.jl/pull/93#discussion_r1328675770

**Comment by yebai** [comment]
@sunxd3, is this actually fixed?

**Comment by sunxd3** [comment]
no, I should've added that this is superseded by https://github.com/TuringLang/JuliaBUGS.jl/issues/327

there are subtle differences, but a good solution of #327 will resolve this issue or makes it much easier
